docs/plugins/integrations/authentication.rst

```diff
@@ -0,0 +1,750 @@
+**************************
+Integration authentication
+**************************
+
+.. contents:: Table of contents
+
+The IntegrationsBundle provides factories and helpers to create Guzzle Client classes for common authentication protocols.
+
+----------
+
+Registering the Integration for authentication
+##############################################
+
+If the Integration requires the User to authenticate through the web (OAuth2 three legged), the Integration needs to tag a service with ``mautic.auth_integration`` to handle the authentication process (redirecting to login, request the access token, etc).
+
+This service needs to implement::
+
+    \Mautic\IntegrationsBundle\Integration\Interfaces\AuthenticationInterface.
+
+.. code-block:: php
+
+    <?php
+    return [
+        // ...
+        'services' => [
+            // ...
+            'integrations' => [
+                // ...
+                'helloworld.integration.authentication' => [
+                    'class' => \MauticPlugin\HelloWorldBundle\Integration\Support\AuthSupport::class,
+                    'tags'  => [
+                        'mautic.auth_integration',
+                    ],
+                ],
+                // ...
+            ],
+            // ...
+        ],
+        // ...
+    ];
+
```

Comment:
Is it appropriate to use the RST markup to explain in more depth some of these examples - like how it is done on these pages:

https://mautic-developer.readthedocs.io/en/latest/plugins/config.html 

What do you think?